FidoCore.md

```diff
@@ -1853,3 +1855,124 @@ begin
   ...
 ```
 
+## Currying
+
+Units: `Fido.Currying`.
+
+Currying allows you to transform a function with multiple parameters in sequence of one-parameter functions.
+
+FidoLib allows you to do that by using the Currying tool:
+
+```pascal
+function Add(const P1: Integer; const P2: Integer): Integer;
+begin
+  Result := P1 + P2;
+end;
+  
+begin
+  Result := Writeln(Curry.Cook<Integer, Integer, Integer>(Add)(1)(2)); //This will return 3
+end;
+```
+
+## Caching
+
+Units: `Fido.caching.Intf`,  `Fido.caching`.
+
+FidoLib support a "functional" caching. That means you can cache the results of functions... basically memoizing.
+
+Since pure memoization is not always optimal, FidoLib supports three types of caching:
+
+- **Memoization**. The cache has no limit.
+- **FIFO**. The cache has a limited size. when the size is reached the first item that was added is removed.
+- **Usage**. The cache has a limited size. when the size is reached the items that are added/accessed the last are kept.
+
+When an item in the cache becomes stale it can be forced by using the `.ForceIt` method.
```

Comment:
It can be forced to be what? From the comment in the code below it can be forced to be recalculated, but that has nothing to do with being stale right?

Reply:
It can be forced with another value.
You know that the result of Whatever(1) is stale, you call Cache.ForceIt(Whatever, 1) to replace the stale value with the current one.